src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -396,6 +398,78 @@ class E2EHyperspaceRulesTests extends HyperspaceSuite with SQLHelper {
       sortedRowsWithHyperspaceDisabled.sameElements(getSortedRows(dfAfterHyperspaceDisabled)))
   }
 
+  test("Verify Join Index Rule utilizes indexes correctly after incremental refresh.") {
+    // Setup. Create Data.
+    val sampleParquetDataLocation = "src/test/resources/sampleparquet"
+    spark.conf.set(IndexConstants.REFRESH_APPEND_ENABLED, true)
+    val refreshTestLocation = sampleParquetDataLocation + "refresh"
+    FileUtils.delete(new Path(refreshTestLocation))
+    val indexConfig = IndexConfig(s"index", Seq("RGUID"), Seq("imprs"))
+    import spark.implicits._
+    SampleData.testData
+      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      .limit(10)
+      .write
+      .parquet(refreshTestLocation)
+    val df = spark.read.load(refreshTestLocation)
+
+    // Create Index.
+    hyperspace.createIndex(df, indexConfig)
+
+    // Append to Original Data.
+    SampleData.testData
+      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      .limit(3)
+      .write
+      .mode("append")
+      .parquet(refreshTestLocation)
+
+    // Refresh Index.
+    hyperspace.refreshIndex(indexConfig.indexName)
+
+    // Create a Join Query
```

Comment:
Is this going to be:

```suggestion
    // Create a Join Query.
```

? :)